app/jobs/regular/emit_web_hook_event.rb

```diff
@@ -74,7 +68,20 @@ def send_webhook!
       end
 
       publish_webhook_event(web_hook_event)
-      retry_web_hook if response&.status != 200
+      process_webhook_response(web_hook_response)
+    end
+
+    def process_webhook_response(web_hook_response)
+      return if web_hook_response&.status.blank?
+
+      case web_hook_response.status
+      when 200..299
+        return
+      when 404, 410
+        web_hook.update_attribute(:active, false) if @retry_count >= MAX_RETRY_COUNT
```

Comment:
I think this is worth a log in in order to keep track of why that webhook was deactivated.

Reply:
Yes, that's actually what I was hoping I get feedback on, so thank you ⭐️ 

What type or kind of log do can I have here? Are we talking about `Rails.logger.info` or we have some other type of logging within Discourse?

Reply:
[StaffActionLog](https://github.com/discourse/discourse/blob/master/app/services/staff_action_logger.rb) is what I would recommend.